src/doc/building_visit/Building_Directly_With_CMake.rst

```diff
@@ -16,3 +16,234 @@ If ``build_visit`` was used to build VisIt_ on the platform in the past, it shou
+
+VISIT_ENGINE_ONLY : BOOL : OFF
+    Toggles building of only the compute engine and its plugins.
+
+VISIT_SERVER_COMPONENTS_ONLY : BOOL : OFF
+    Build only vcl, mdserver, engine and their plugins.
+
+VISIT_ENABLE_LIBSIM : BOOL : ON
+    Toggles building of libsim.
+
+
+Controlling plugins being built
+"""""""""""""""""""""""""""""""
+
+VISIT_BUILD_ALL_PLUGINS : BOOL : OFF
+    Toggles the building of all plugins.
+    When turned on the following optional plugins will be added to the build:
+
+    Database: PICS_Tester, Rect
+
+    Operator: Context ConnCompReduce, MetricThreshold, RemoveCells, SurfCompPrep
+
+    Plot: Topology
+
+    Note: the list of optional plugins is subject to change.
+
+VISIT_BUILD_MINIMAL_PLUGINS : BOOL : OFF
+    Toggles the building of a minimal set of database, operator, and plot plugins.
+    When turned on, only the following plugins will be built:
+
+    Database: Curve2D, RAW, VTK, PICS_Tester
+
+    Operator: Lineout, Slice, Threshold
+
+    Plot: Curve, Mesh, Pseudocolor
+
+    Note: the list of minimal plugins is subject to change.
+
+VISIT_SELECTED_DATABASE_PLUGINS : STRING
+    ';' separated list of database plugins to build, eg: VTK;Silo
+
+     If not empty, will supercede the settings of VISIT_BUILD_MINIMAL_PLUGINS and VISIT_BUILD_ALL_PLUGINS for database plugins.
+
+VISIT_SELECTED_OPERATOR_PLUGINS : STRING
+    ';' separated list of operator plugins to build, eg: Slice;Lineout;Transform
+
+     If not empty, will supercede the settings of VISIT_BUILD_MINIMAL_PLUGINS and VISIT_BUILD_ALL_PLUGINS for operator plugins.
+
+VISIT_SELECTED_PLOT_PLUGINS : STRING
+    ';' separated list of plot plugins to build, eg: Mesh;Pseudocolor
+
+     If not empty, will supercede the settings of VISIT_BUILD_MINIMAL_PLUGINS and VISIT_BUILD_ALL_PLUGINS for plot plugins.
+
+
+Controlling extra tools being built
+"""""""""""""""""""""""""""""""""""
+VISIT_ENABLE_ANNOTATION_TOOLS : BOOL : ON
+    Toggles the generation of annotation tools: text2polys, time_annotation.
+
+VISIT_ENABLE_DATAGEN : BOOL: ON
+    Toggles the generation of sample data files.
+
+VISIT_ENABLE_DATA_MANUAL_EXAMPLES: BOOL : OFF
+    Toggles generation of :ref:`Getting Data Into Visit<Getting Data into VisIt_>` examples.
+
+VISIT_ENABLE_DIAGNOSTICS : BOOL : ON
+    Toggles building of diagnostic tools: exceptiontest, mpitest, networktest, osmesatest.
+
+VISIT_ENABLE_MANUALS : BOOL : ON
+    Toggles building of manuals, requires Sphinx in Python.
+
+VISIT_ENABLE_SILO_TOOLS : BOOL : ON
+    Toggles building of Silo tools: mrgtree2dot, add_visit_searchpath.
+
+VISIT_ENABLE_UNIT_TESTS : BOOL : ON
+    Toggles building of unit tests: MRUCache, Namescheme, Utilty, StringHelpers, exprconfig, exprtest.
+
+Useful for developers
+"""""""""""""""""""""
+
+
+VISIT_CREATE_SOCKET_RELAY_EXECUTABLE: BOOL : ON
+    Toggles creation of separate executble that forwards VisIt's socket conneciton between engine and component launcher.
```

Comment:
Typo "conneciton" to "connection".

Reply:
@brugger1 I think its almost easier to use the [propose change feature -- step 7 here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request) and do this...

```suggestion
    Toggles creation of separate executble that forwards VisIt's socket connection between engine and component launcher.
```

Reply:
done